src/averageAges.js

```diff
@@ -55,10 +71,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const mothers = people.filter(person => {
+    return people.some(item => item.mother === person.name);
+  });
+
+  const peopleWithMother = people.filter(person => {
+    return onlyWithSon !== undefined
+      ? mothers.find(mother => mother.name === person.mother)
+    && person.sex === 'm'
+      : mothers.find(mother => mother.name === person.mother);
+  });
+
+  const totalAgeDiff = peopleWithMother.reduce((sum, person) => {
+    const mother = mothers.find(item => item.name === person.mother);
+
+    return sum + (person.born - mother.born);
+  }, 0);
+
+  return +(totalAgeDiff / peopleWithMother.length).toFixed(2);
```

Comment:
Same as above, replace `+(totalAgeDiff / peopleWithMother.length).toFixed(2);` with `Number((totalAgeDiff / peopleWithMother.length).toFixed(2));` for better readability.